File: queues using stacks.py
class MyQueue:

    def __init__(self):
        #creating two stacks because we will be using one to store and one to pop for better complexity
        self.instack = []
        self.outstack = []

    def push(self, x: int) -> None:
        self.instack.append(x)    #just inserting the data into instack    TC: O(1) SC:O(1)

    def pop(self) -> int:
        self.peek()
        return self.outstack.pop()      #popping the items is 0(1) for time and space compelxity

    def peek(self) -> int:                      #just looking at peek of the outstack
        if not self.outstack:                   #if the outstack is empty. it goes through the next while loop to add items from instack to outstack
            while self.instack:                         #time complexity is O(n) space compexity is o(1)
                self.outstack.append(self.instack.pop())
        return self.outstack[-1]

    def empty(self) -> bool:                   # if the both the stacks are empty it returns true else false
        return not self.instack and not self.outstack

File: QueuesUsingStacks.cs
public class MyQueue
    {

        // Time Complexity : Push - O(1), Pop - O(1), Peek - O(1) and Empty - O(1)
        // Space Complexity : O(1)
        // Did this code successfully run on Leetcode : Yes
        // Any problem you faced while coding this : No

        /* Approach
         * Use 2 stacks one for pushing incoming elements and one to pop from
         * Keep pushing elements to instack until we get a pop or a peek
         * once we get pop or peek, we then move all elements from instack to outstack and pop the top or peek the top
         * */

        Stack<int> inStack;
        Stack<int> outStack;


        public MyQueue()
        {
            inStack = new Stack<int>();
            outStack = new Stack<int>();
        }

        public void Push(int x)
        {
            inStack.Push(x);
        }

        public int Pop()
        {
            Peek();
            return outStack.Pop();
        }

        public int Peek()
        {
            if(outStack.Count == 0)
            {
                while(inStack.Count > 0)
                {
                    outStack.Push(inStack.Pop());
                }
            }
            return outStack.Peek();
        }

        public bool Empty()
        {
            return inStack.Count == 0 && outStack.Count == 0;
        }
    }
